packages/builders/src/util/ValidationError.ts

```diff
@@ -0,0 +1,21 @@
+import { z } from 'zod/v4';
+
+/**
+ * An error that is thrown when validation fails.
+ */
+export class ValidationError extends Error {
+	/**
+	 * The underlying cause of the validation error.
+	 */
+	public override readonly cause: z.ZodError;
+
+	/**
+	 * @internal
+	 */
+	public constructor(error: z.ZodError) {
```

Comment:
Should this be `private` then?

Reply:
You could make the constructor private/protected, but then you'd need to hack around it with `Reflect.construct()` calls. Imo it's fine if the constructor is public, if someone wants to build the class for testing or w/e they should be able to